app/src/main/java/com/nutomic/syncthingandroid/util/Util.java

```diff
@@ -172,6 +171,38 @@ public static boolean nativeBinaryCanWriteToPath(Context context, String absolut
         return true;
     }
 
+    /**
+     * Returns if the syncthing binary would be able to write a file into
+     * the given folder given the configured access level.
+     *
+     * This uses the Android-native File API.
+     */
+    public static Boolean nativeBinaryCanWriteToPath2(File externalStorageDir, String absoluteFolderPath) {
+        final String TOUCH_FILE_NAME = ".stwritetest";
+
+        // Normalize path replacing ~ with external storage directory.
+        // This is consistent with what $HOME is set to in SyncthingRunnable.
+        final String normalizedPath = absoluteFolderPath.replaceFirst("^~", externalStorageDir.getAbsolutePath());
```

Comment:
Did you test explicitly if this part alone helped?

Seems like we should use the same environment, including setting `HOME`, when executing the shell command as we do in SyncthingRunnable.